homeassistant/components/iammeter/config_flow.py

```diff
@@ -0,0 +1,123 @@
+"""Config flow for iammeter integration."""
+import asyncio
+import logging
+from urllib.parse import ParseResult, urlparse
+
+import async_timeout
+import iammeter
+from iammeter.power_meter import IamMeterError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT
+from homeassistant.core import HomeAssistant, callback
+
+from .const import DEFAULT_HOST, DEFAULT_NAME, DEFAULT_PORT, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_TIMEOUT = 8
+
+
+@callback
+def iammeter_entries(hass: HomeAssistant):
+    """Return the hosts already configured."""
+    return {
+        entry.data[CONF_NAME] for entry in hass.config_entries.async_entries(DOMAIN)
+    }
+
+
+class IammeterConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Iammeter."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self.host = None
+        self._errors = {}
+        self.discovered_conf = {}
+
+    def _host_in_configuration_exists(self, host) -> bool:
+        """Return True if host exists in configuration."""
+        if host in iammeter_entries(self.hass):
+            return True
+        return False
+
+    async def _test_connection(self, host, port):
+        try:
+            with async_timeout.timeout(PLATFORM_TIMEOUT):
+                await iammeter.real_time_api(host, port)
+                return True
+        except (IamMeterError, asyncio.TimeoutError):
+            _LOGGER.error("Device is not ready!")
+            self._errors[CONF_NAME] = "cannot_connect"
```

Comment:
Raise an exception instead. Please use our scaffold script for best practices code.